src/header/common/link.rs

```diff
@@ -0,0 +1,975 @@
+        }
+        if let Some(ref href_lang) = self.href_lang {
+            for tag in href_lang {
+                try!(write!(f, "; hreflang={}", tag));
+            }
+        }
+        if let Some(ref media_desc) = self.media_desc {
+            try!(fmt_delimited(f, media_desc.as_slice(), ", ", "; media=\"", "\""));
+        }
+        if let Some(ref title) = self.title {
+            try!(write!(f, "; title=\"{}\"", title));
+        }
+        if let Some(ref title_star) = self.title_star {
+            try!(write!(f, "; title*={}", title_star));
+        }
+        if let Some(ref media_type) = self.media_type {
+            try!(write!(f, "; type=\"{}\"", media_type));
+        }
+        if let Some(ref link_extension) = self.link_extension {
+            try!(write!(f, "; link-extension={}", link_extension));
+        }
+
+        Ok(())
+    }
+}
+
+impl FromStr for Link {
+    type Err = ::Error;
+
+    fn from_str(s: &str) -> ::Result<Link> {
+        // Create a split iterator with delimiters: `;`, `,`
+        let link_split = SplitAsciiUnquoted::new(s, ";,");
+
+        let mut link = Link::new(Vec::new());
+
+        // Loop over the splits parsing the Link header into
+        // a `Vec<LinkValue>`
+        for segment in link_split {
+            // Parse the `Target IRI`
+            // https://tools.ietf.org/html/rfc5988#section-5.1
+            if segment.trim().starts_with('<') {
+                link.values.push(
+                    match verify_and_trim(segment.trim(), b'<', b'>') {
+                        Err(_) => return Err(::Error::Header),
+                        Ok(link_url) => match Uri::new(link_url) {
+                            Err(_) => return Err(::Error::Header),
+                            Ok(uri) => LinkValue {
+                                link: uri,
+                                rel: None,
+                                anchor: None,
+                                rev: None,
+                                href_lang: None,
+                                media_desc: None,
+                                title: None,
+                                title_star: None,
+                                media_type: None,
+                                link_extension: None,
+                            }
+                        },
+                    }
+                );
+            } else {
+                // Parse the current link-value's parameters
+                let mut link_param_split = segment.splitn(2, '=');
+
+                let link_param_name = match link_param_split.next() {
+                    None => return Err(::Error::Header),
+                    Some(p) => p.trim(),
+                };
+
+                let mut link_header = match link.values.last_mut() {
+                    None => return Err(::Error::Header),
+                    Some(l) => l,
+                };
+
+                if "rel".eq_ignore_ascii_case(link_param_name) {
```

Comment:
When we want case-insensitive comparison I am using this function since we're dealing only with ASCII.